ui/components/app/permission-page-container/permission-page-container.component.js

```diff
@@ -103,6 +104,17 @@ export default class PermissionPageContainer extends Component {
 
     if (Object.keys(request.permissions).length > 0) {
       approvePermissionsRequest(request);
+      const searchParams = new URLSearchParams(
+        this.props.history.location.search,
+      );
+      const redirect = searchParams.get('redirect');
+      if (redirect) {
+        window.open(
+          `${redirect}?${request.approvedAccounts
+            .map((address) => `approvedAccounts=${address}`)
```

Comment:
Don't we want to encode the query part? 
```
.map((address) => `approvedAccounts=${encodeURIComponent(address)}`)
```

Reply:
I think account addresses will be uri safe but I can make this change to cover all bases